genie/clinical.py

```diff
@@ -14,22 +14,65 @@
 logger = logging.getLogger(__name__)
 
 
-def remove_greaterthan_lessthan_str(col):
-    '''
-    In clinical file, there are redacted value such as >89 and <17.
-    These < and > signs must be removed
-    '''
-    try:
-        col = [
-            text.replace(">", "")
-            if isinstance(text, str) else text for text in col]
-        col = [
-            int(text.replace("<", ""))
-            if isinstance(text, str) and text != "" else text
-            for text in col]
-    except ValueError:
-        pass
-    return(col)
+# def remove_greaterthan_lessthan_str(col):
+#     '''
+#     In clinical file, there are redacted value such as >89 and <17.
+#     These < and > signs must be removed
+#     '''
+#     try:
+#         col = [
+#             text.replace(">", "")
+#             if isinstance(text, str) else text for text in col]
+#         col = [
+#             int(text.replace("<", ""))
+#             if isinstance(text, str) and text != "" else text
+#             for text in col]
+#     except ValueError:
+#         pass
+#     return(col)
+def _check_year(clinicaldf: pd.DataFrame, year_col: int, filename: str,
+                allowed_string_values: list = []) -> str:
+    """Check year columns
+
+    Args:
+        clinicaldf: Clinical dataframe
+        year_col: YEAR column
+        filename: Name of file
+        allowed_string_values: list of other allowed string values
+
+    Returns:
+        Error message
+    """
+    error = ''
+    if process_functions.checkColExist(clinicaldf, year_col):
+        # Deal with pre-redacted values and other allowed strings
+        # first because can't int(text) because there are
+        # instances that have <YYYY
+        year_series = clinicaldf[year_col][
+            ~clinicaldf[year_col].isin(allowed_string_values)
+        ]
+        year_now = datetime.datetime.utcnow().year
```

Comment:
Remove this validation

Reply:
Actually looking into this now. None of the YEARS can be greater than the current year, so I will not remove this.